Framework/Muon/src/MuonPreProcess.cpp

```diff
@@ -214,21 +241,80 @@ MatrixWorkspace_sptr MuonPreProcess::applyTimeOffset(MatrixWorkspace_sptr ws,
   }
 }
 
+MatrixWorkspace_sptr
+MuonPreProcess::applyTimeZeroTable(MatrixWorkspace_sptr ws,
+                                   const TableWorkspace_sptr &timeZeroTable) {
+  auto cloneWs = cloneWorkspace(ws);
+  if (!timeZeroTable) {
+    return cloneWs;
+  }
+  const auto numSpec = cloneWs->getNumberHistograms();
+  for (auto specNum = 0u; specNum < numSpec; ++specNum) {
+    auto &xData = cloneWs->mutableX(specNum);
+    for (auto &xValue : xData) {
+      API::TableRow row = timeZeroTable->getRow(specNum);
+      xValue -= row.Double(0);
+    }
+  }
+  return cloneWs;
+}
+
 MatrixWorkspace_sptr MuonPreProcess::applyCropping(MatrixWorkspace_sptr ws,
                                                    const double &xMin,
                                                    const double &xMax) {
-  if (xMin != EMPTY_DBL() || xMax != EMPTY_DBL()) {
-    IAlgorithm_sptr crop = createChildAlgorithm("CropWorkspace");
-    crop->setProperty("InputWorkspace", ws);
-    if (xMin != EMPTY_DBL())
-      crop->setProperty("Xmin", xMin);
-    if (xMax != EMPTY_DBL())
-      crop->setProperty("Xmax", xMax);
-    crop->execute();
-    return crop->getProperty("OutputWorkspace");
-  } else {
+  if (xMin == EMPTY_DBL() && xMax == EMPTY_DBL())
     return ws;
+
+  if (getPropertyValue("TimeZeroTable").empty())
+    return cropWithSingleValues(ws, xMin, xMax);
+  else
+    return cropWithVectors(ws, xMin, xMax);
+}
+
+MatrixWorkspace_sptr
+MuonPreProcess::cropWithSingleValues(MatrixWorkspace_sptr ws,
+                                     const double &xMin, const double &xMax) {
```

Comment:
One small nit - it's good practice to not pass primitives by const ref. 
Just rely on a copy as it costs the same, has the same safety semantics and guarantees no "spooky action at a distance" so the compiler can better optimise

See [this SO answer for some further reading](https://stackoverflow.com/a/17133906)

Reply:
Oops that's my mistake. My original thought was to modify them within a function but I decided to do it differently and forgot to change the function definitions. Thanks for spotting that.